agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/SecondEntryPoint.java

```diff
@@ -562,12 +571,15 @@ private static SdkMeterProviderBuilder configureMetrics(
     MetricDataMapper mapper =
         new MetricDataMapper(
             telemetryClient::populateDefaults, configuration.preview.captureHttpServer4xxAsError);
-    metricReader =
+    PeriodicMetricReaderBuilder readerBuilder =
         PeriodicMetricReader.builder(
-                new AgentMetricExporter(
-                    metricFilters, mapper, telemetryClient.getMetricsBatchItemProcessor()))
-            .setInterval(Duration.ofSeconds(configuration.preview.metricIntervalSeconds))
-            .build();
+            new AgentMetricExporter(
+                metricFilters, mapper, telemetryClient.getMetricsBatchItemProcessor()));
+    int intervalMillis =
+        Integer.getInteger(
+            "applicationinsights.testing.metric-reader-interval-millis",
```

Comment:
can we add comment to indicate that this is for testing purpose only?

Reply:
all properties under `applicationinsights.testing.*` are for testing purposes only